lib/hook.js

```diff
@@ -174,6 +175,39 @@ function unhookRunInThisContext() {
     vm.runInThisContext = originalRunInThisContext;
 }
 
+/**
+ * hooks `vm.runInNewContext` to return transformed code.
+ * @method hookRunInNewContext
+ * @static
+ * @param matcher {Function(filePath)} a function that is called with the filename passed to `vm.runInNewContext`
+ *  Should return a truthy value when transformations need to be applied to the code, a falsy value otherwise
+ * @param transformer {Function(code, filePath)} a function called with the original code and the filename passed to
+ *  `vm.createScript`. Should return the transformed code.
+ * @param options {Object} options Optional.
+ * @param {Boolean} [options.verbose] write a line to standard error every time the transformer is called
+ */
+function hookRunInNewContext(matcher, transformer, opts) {
+    opts = opts || {};
+    var fn = transformFn(matcher, transformer, opts.verbose);
+    vm.runInNewContext = function (code, context, file) {
+        var ret = fn(code, file);
+        Object.keys(global).forEach(function(key) {
+          if (key.indexOf('$$cov_') === 0) {
```

Comment:
You only need to do all these key shennanigans if you are trying to figure out what the coverage variable is inside the hook module (which we are taking out). At the point where the caller registers the hook, it knows _exactly_ what the coverage variable is.

See:
https://github.com/laurentgoudet/istanbul/blob/6d17b04d0b106e80c7a9a9c9dd4218eb0e104a86/lib/command/common/run-with-cover.js#L149
https://github.com/laurentgoudet/istanbul/blob/6d17b04d0b106e80c7a9a9c9dd4218eb0e104a86/lib/command/common/run-with-cover.js#L188

(you might need to initialize the coverage variable early before registering the hook so you can reference it)